TddShop/TddShop.Cli.Tests/Shipment/AncientRomeShippingServiceTests.cs

```diff
@@ -13,13 +14,184 @@ namespace TddShop.Cli.Tests.Shipment
     public class AncientRomeShippingServiceTests
     {
         private Mock<IDeliveryService> _deliveryService;
+        private Mock<INumeralsConvereter> _numeralsConvereter;
         private AncientRomeShippingService _target;
 
         [SetUp]
         public void Initialize()
         {
             _deliveryService = new Mock<IDeliveryService>();
-            _target = new AncientRomeShippingService(_deliveryService.Object);
+            _numeralsConvereter = new Mock<INumeralsConvereter>();
+            _target = new AncientRomeShippingService(_deliveryService.Object, _numeralsConvereter.Object);           
         }
+        [Test]
+        public void ShipOrder_IfOrderContainsAtLeast1item_DeliveryServiceShouldReturnShipmentReferenceNumberOnce()
+        {
+            //Arrange            
+             var order = new OrderModel
+             {
+                 Items = new ItemModel[] {
+                    new ItemModel
+                    {
+                        Name = "przedmiot",
+                        Quantity = 2
+                    }
+
+                }
+             };
+            var itemsNumber = 1;            
+
+            //Act
+            _target.ShipOrder(order);
+
+            //Assert
+            _deliveryService.Verify(x => x.GenerateShipmentReferenceNumber(itemsNumber), Times.Once);
+
+        }
+
+        [Test]
+        public void ShipOrder_IfOrderIsEmpty_DeliveryServiceShouldNeverReturnShipmentReferenceNumber()
+        {
+            //Arrange            
+            var order = new OrderModel();
+            var itemsNumber = 0;
+
+            //Act
+            _target.ShipOrder(order);
+
+            //Assert
+            _deliveryService.Verify(x => x.GenerateShipmentReferenceNumber(itemsNumber), Times.Never);
+
+        }
+
+        [Test]
+        public void ShipOrder_IfShipmentReferenceNumberGraterThen3999_ArabicToRomanConverterShouldNeverConvert()
```

Comment:
Tutaj lepiej by było sprawdzić, czy numer przesyłki się nie wygeneruje w takim razie. Tak samo order - nigdy nie powinien być wysłany. To jest bardziej istotne niż sprawdzenie, czy został wywołany konwerter liczb.